src/requestHandler.ts

```diff
@@ -383,8 +386,12 @@ export default class RequestHandler {
       return;
     }
 
+    const directory = path.relative(process.cwd(), path.dirname(filepath))
+    if (!await this.app.vault.adapter.exists(directory)) await this.app.vault.adapter.mkdir(directory)
+    await this.app.vault.adapter.write(filepath, req.body);
```

Comment:
Since we're doing this twice, we should probably consider wrapping lines 389:391 into their own function that we can call from multiple locations.